extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/MongoDbConnection.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.jet.mongodb;
+
+import com.hazelcast.function.ConsumerEx;
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.JetException;
+import com.hazelcast.jet.retry.RetryStrategies;
+import com.hazelcast.jet.retry.RetryStrategy;
+import com.hazelcast.jet.retry.impl.RetryTracker;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.mongodb.client.MongoClient;
+
+import static com.hazelcast.internal.util.Preconditions.checkState;
+import static com.hazelcast.jet.retry.IntervalFunction.exponentialBackoffWithCap;
+
+/***
+ * Manages connection to MongoDB, reconnects if necessary.
+ */
+class MongoDbConnection {
+    @SuppressWarnings("checkstyle:MagicNumber")
+    private static final RetryStrategy RETRY_STRATEGY = RetryStrategies.custom()
+                                                                       .intervalFunction(exponentialBackoffWithCap(100
+                                                                               , 2.0, 3000))
+                                                                       .maxAttempts(20)
+                                                                       .build();
```

Comment:
Doesn't MongoDB driver have its own retry mechanism? Is there a scenario where we can't use it?

Reply:
Simple operations are retried. In case of e.g. Transaction Transient Error we have to retry manually